src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,81 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { useAppDispatch } from '../../app/hook';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+
+interface Props {
+  todo: Todo;
+}
+
+export const TodoModal: React.FC<Props> = ({ todo }) => {
+  const dispatch = useAppDispatch();
+
+  const clear = () => {
+    dispatch(currentTodoActions.clearTodo());
+  };
+
+  const [user, setUser] = useState<User>();
+  const [loader, setLoader] = useState(false);
+
+  useEffect(() => {
+    setLoader(true);
+
+    getUser(todo.userId)
+      .then(setUser)
+      .finally(() => setLoader(false));
```

Comment:
Consider adding error handling in the promise chain of `getUser` to manage potential errors during the API call. This can be done by adding a `.catch()` method to handle any errors that occur.